src/network/receive.ts

```diff
@@ -496,7 +497,8 @@ export async function parseKeysendInvoice(
   i: interfaces.Invoice
 ): Promise<void> {
   try {
-    const hash = i.r_hash.toString('base64')
+    // const hash = i.r_hash.toString('base64')
+    const hash = sha.sha256.hex(JSON.stringify(i))
```

Comment:
@antonilol so this caching system was added for greenlight because for some reason it was showing duplicate messages, the PR and the issue is below

https://github.com/stakwork/sphinx-relay/issues/274
https://github.com/stakwork/sphinx-relay/pull/277

Reply:
duplicate